### PR TITLE
Story4

### PR DESCRIPTION
**Pull request by Richtbiscuitcoder**
https://trello.com/c/Y8mTOpYT/71-story-4-add-new-questions

please review our code :)

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -11,6 +11,6 @@
 Route::get('/quizzes', [QuizAPIController::class, 'displayAllQuizzes']);
 Route::get('/quizzes/{id}', [QuizAPIController::class, 'displayQuizByID']);
 Route::post('/quizzes', [QuizAPIController::class, 'addNewQuiz']);
-
+Route::post('/questions', [QuizAPIController::class, 'addNewQuestion']);
```
Comment (mallowmew) [inline routes/api.php]: Do requests to the `/questions` route go in the Quiz controller?

File: tests/Feature/QuizTest.php [hunk 2]
```diff
@@ -133,4 +134,82 @@ public function test_getQuizByID_success(): void
...
+        $response = $this->postJson('/api/questions', $testData);
+
+        $response->assertStatus(201)
+            ->assertJson(function (AssertableJson $json) {
+                $json->hasAll(['message']);
+            });
+        $this->assertDatabaseHas('questions', $testData);
+    }
+
+    public function test_addNewQuestion_questionValidation(): void
```
Comment (mallowmew) [inline tests/Feature/QuizTest.php]:
```suggestion
    public function test_addNewQuestion_missingQuestion(): void
```

File: tests/Feature/QuizTest.php [hunk 3]
```diff
@@ -133,4 +134,82 @@ public function test_getQuizByID_success(): void
...
+        $response->assertStatus(422)
+            ->assertJson(function (AssertableJson $json) {
+                $json->hasAll(['message', 'errors'])
+                    ->has('errors', function (AssertableJson $json) {
+                        $json->hasAll('question');
+                    });
+            });
+    }
+
+    public function test_addNewQuestion_pointsValidation(): void
```
Comment (mallowmew) [inline tests/Feature/QuizTest.php]:
```suggestion
    public function test_addNewQuestion_missingPoints(): void
```

**Review by mallowmew — CHANGES_REQUESTED** [review]

File: tests/Feature/QuizTest.php
Comment (mallowmew) [inline tests/Feature/QuizTest.php:1]: Can you think of a test case that would cause a failure to save into the database?

**Review by mallowmew — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/QuestionAPIController.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+use Illuminate\Http\Request;
+
+class QuestionAPIController extends Controller
+{
+    public function addNewQuestion( Request $request): JsonResponse
+    {
+        $request->validate([
+            'question' => 'required|string|min:1|max:255',
+            'hint' => 'string|max:255',
+            'points' => 'required|integer',
```
Comment (mallowmew) [inline app/Http/Controllers/QuestionAPIController.php]: Is 0 points acceptable? -2 points?

File: tests/Feature/QuizTest.php
Comment (Richtbiscuitcoder) [reply tests/Feature/QuizTest.php:1]: no..

**Review by mallowmew — APPROVED** [review]

File: app/Http/Controllers/QuestionAPIController.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+use Illuminate\Http\Request;
+
+class QuestionAPIController extends Controller
+{
+    public function addNewQuestion( Request $request): JsonResponse
+    {
+        $request->validate([
+            'question' => 'required|string|min:1|max:255',
+            'hint' => 'string|max:255',
+            'points' => 'required|min:1|integer',
```
Comment (mallowmew) [inline app/Http/Controllers/QuestionAPIController.php:17]: Hold on - what if you try to add a question for a quiz that doesn't exist? Should this be allowed?

**Review by mallowmew — CHANGES_REQUESTED** [review]

**Review by mallowmew — APPROVED** [review]

**Pull request merged**